gamla/functional.py

```diff
@@ -476,6 +476,20 @@ def apply_inner(function):
     return apply_inner
 
 
+def apply_async(*args, **kwargs):
```

Comment:
consider moving things to their own files rather than expanding `functional.py` (same for the tests)